app/controllers/api/v1/book_controller.rb

```diff
@@ -5,6 +5,19 @@ def index
         render_paginated Book.all
       end
 
+      def isbn
+        begin
+          isbn = params.require(:isbn)
```

Comment:
Lot of better, one thing to improve this, try to rescue the error in your application controller, because this kind of error occurs often. Like you did for Pundit:
`rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized`